curvefs/src/client/s3/client_s3_cache_manager.h

```diff
@@ -85,112 +85,146 @@ struct ObjectChunkInfo {
     uint64_t objectOffset;  // s3 object's begin in the block
 };
 
+struct PageData {
+    uint64_t index;
+    char* data;
+};
+using PageDataMap = std::map<uint64_t, PageData*>;
+
 class DataCache {
  public:
-    DataCache(S3ClientAdaptorImpl *s3ClientAdaptor,
+    DataCache(S3ClientAdaptorImpl* s3ClientAdaptor,
+              ChunkCacheManager* chunkCacheManager, uint64_t chunkPos,
+              uint64_t len, const char* data);
+    /*DataCache(S3ClientAdaptorImpl *s3ClientAdaptor,
               ChunkCacheManager *chunkCacheManager, uint64_t chunkPos,
-              uint64_t len, const char *data)
-        : s3ClientAdaptor_(s3ClientAdaptor),
-          chunkCacheManager_(chunkCacheManager), chunkPos_(chunkPos), len_(len),
-          dirty_(true) {
-        data_ = new char[len];
-        memcpy(data_, data, len);
-        createTime_ = ::curve::common::TimeUtility::GetTimeofDaySec();
-        //  dirty_.exchange(true, std::memory_order_acq_rel);
+              uint64_t len);*/
+    virtual ~DataCache() {
+        auto iter = dataMap_.begin();
+        for (; iter != dataMap_.end(); iter++) {
+            auto pageIter = iter->second.begin();
+            for (; pageIter != iter->second.end(); pageIter++) {
+                delete pageIter->second->data;
```

Comment:
Can delete in here and ReleasePageData concurrntly will cause an concurrency problem？